lib/hw_unique_key/hw_unique_key.c

```diff
@@ -4,58 +4,270 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+#include <string.h>
+#include <nrf_cc3xx_platform_kmu.h>
+#include <nrf_cc3xx_platform_ctr_drbg.h>
+#include <hw_unique_key.h>
+#include <nrfx_nvmc.h>
+
+#ifdef __ZEPHYR__
 #include <sys/printk.h>
+#include <kernel.h>
+#define HUK_PRINT printk
+#define HUK_PANIC k_panic
+
+#elif defined(__NRF_TFM__)
+#include "log/tfm_log.h"
+#include "utilities.h"
+#define HUK_PRINT LOG_MSG
+#define HUK_PANIC tfm_core_panic
+
+#else
+static void panic(void)
+{
+	while (1)
+		;
+}
+#define HUK_PRINT(...)
+#define HUK_PANIC panic
+
+#endif /* __ZEPHYR__ */
+
+
+#if !HUK_HAS_KMU
 #include <pm_config.h>
-#include <string.h>
 #include <fprotect.h>
-#include <init.h>
-
 #include <hal/nrf_acl.h>
-#include <hal/nrf_ficr.h>
-
-/** The nrf52840 doesn't have a KMU but the helper functions
- * handling the KDR key are inside the following header
- */
-#include <nrf_cc3xx_platform_kmu.h>
 
 /* The magic data identify the flash page which contains the key */
-static uint8_t huk_magic[16] = {
-	0x56, 0xB5, 0x6D, 0xAD, 0xAB, 0x5B, 0x6A, 0xB5,
-	0x55, 0xAB, 0x5A, 0xDA, 0xAB, 0x55, 0xAD, 0x5A
+static const uint32_t huk_magic[4] = {
+	0xAD6DB556, 0xB56A5BAB, 0xDA5AAB55, 0x5AAD55AB
 };
 
 static uint32_t huk_addr = PM_HW_UNIQUE_KEY_PARTITION_ADDRESS;
 
-/* Load the Hardware Unique Key stored in flash */
-int hw_unique_key_load(const struct device *unused)
+void hw_unique_key_write(uint32_t kmu_slot, const uint8_t *key)
+{
+	if (kmu_slot != HUK_KEYSLOT_KDR) {
+		HUK_PRINT("KMU slot must be HUK_KEYSLOT_KDR on nRF52840\n\r");
+		HUK_PANIC();
+	}
+
+	nrfx_nvmc_words_write(huk_addr, huk_magic, sizeof(huk_magic)/4);
+	while (!nrfx_nvmc_write_done_check())
+		;
+	nrfx_nvmc_words_write(huk_addr + sizeof(huk_magic), key, HUK_SIZE_WORDS);
+	while (!nrfx_nvmc_write_done_check())
+		;
+}
+
+bool hw_unique_key_is_written(uint32_t kmu_slot)
 {
-	ARG_UNUSED(unused);
+	if (kmu_slot != HUK_KEYSLOT_KDR) {
+		HUK_PRINT("KMU slot must be HUK_KEYSLOT_KDR on nRF52840\n\r");
```

Comment:
Again, only supported for nrf52840? ANd also why have `kmu_slot` as argument if only one value is allowed?